### ISSUE TITLE
Postgres Error: (invalid_column_reference) after RDS/Postgres Space added

### ISSUE THREAD

**Issue by royki** [blockscout/blockscout#4997]
*Describe your issue here.*
We are running Blockscout Explorer in Volta/EWC network with Open Ethereum archive node which is fully synced. Due to space shortage in RDS, volta explorer was down for sometimes. When the RDS space was added and explorer restarted. We started receiving errors from RDS. One error disappears as time progress like 
```
blockscout_1  | ** (Postgrex.Error) ERROR 23503 (foreign_key_violation) insert or update on table "internal_transactions" violates foreign key constraint "internal_transactions_transaction_hash_fkey"
blockscout_1  | 
blockscout_1  |     table: internal_transactions
blockscout_1  |     constraint: internal_transactions_transaction_hash_fkey
blockscout_1  | 
blockscout_1  | Key (transaction_hash)=(\x091e8ec6422b5489d8302842cd67c6b201966a729fb8ab8c671913341ddcdbfc) is not present in table "transactions".
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:751: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:660: Ecto.Adapters.SQL.insert_all/8
blockscout_1  |     (ecto 3.5.5) lib/ecto/repo/schema.ex:54: Ecto.Repo.Schema.do_insert_all/6
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/internal_transactions.ex:148: Explorer.Chain.Import.Runner.InternalTransactions.insert/3
blockscout_1  |     (ecto 3.5.5) lib/ecto/multi.ex:646: Ecto.Multi.apply_operation/5
blockscout_1  | Function: &Indexer.BufferedTask.log_run/1
``` 

But the following error still persists 
**(Postgrex.Error) ERROR 42P10 (invalid_column_reference) there is no unique or exclusion constraint matching the ON CONFLICT specification**
```
blockscout_1  | Retrying.
blockscout_1  | 2021-12-09T15:35:43.359 application=explorer fetcher=block_catchup first_block_number=14818575 last_block_number=14818575 import_id=1259216934 transaction_id=1259217062 [error] Could not insert all of chunk into Elixir.Explorer.Chain.Address.TokenBalance using options because of error.
blockscout_1  | 
blockscout_1  | Chunk Size: 1
blockscout_1  | Chunk:
blockscout_1  | 
blockscout_1  | [%{address_hash: %Explorer.Chain.Hash{byte_count: 20, bytes: <<45, 8, 82, 190, 53, 168, 180, 228, 255, 126, 136, 214, 157, 158, 154, 191, 152, 133, 155, 125>>}, block_number: 14818575, inserted_at: ~U[2021-12-09 15:35:43.324915Z], token_contract_address_hash: %Explorer.Chain.Hash{byte_count: 20, bytes: <<147, 81, 112, 134, 35, 99, 89, 215, 106, 154, 82, 12, 117, 122, 247, 19, 248, 96, 53, 107>>}, updated_at: ~U[2021-12-09 15:35:43.324915Z]}]
blockscout_1  | 
blockscout_1  | Options:
blockscout_1  | 
blockscout_1  | [conflict_target: [:address_hash, :token_contract_address_hash, :block_number], on_conflict: #Ecto.Query<from t0 in Explorer.Chain.Address.TokenBalance, where: fragment("EXCLUDED.value IS NOT NULL") and (is_nil(t0.value_fetched_at) or fragment("? < EXCLUDED.value_fetched_at", t0.value_fetched_at)), update: [set: [value: fragment("EXCLUDED.value"), value_fetched_at: fragment("EXCLUDED.value_fetched_at"), inserted_at: fragment("LEAST(EXCLUDED.inserted_at, ?)", t0.inserted_at), updated_at: fragment("GREATEST(EXCLUDED.updated_at, ?)", t0.updated_at)]]>, returning: true, timeout: 60000, timestamps: %{inserted_at: ~U[2021-12-09 15:35:43.324915Z], updated_at: ~U[2021-12-09 15:35:43.324915Z]}]
blockscout_1  | 
blockscout_1  | Exception:
blockscout_1  | 
blockscout_1  | ** (Postgrex.Error) ERROR 42P10 (invalid_column_reference) there is no unique or exclusion constraint matching the ON CONFLICT specification
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:751: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:660: Ecto.Adapters.SQL.insert_all/8
blockscout_1  |     (ecto 3.5.5) lib/ecto/repo/schema.ex:54: Ecto.Repo.Schema.do_insert_all/6
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/address/token_balances.ex:67: Explorer.Chain.Import.Runner.Address.TokenBalances.insert/3
blockscout_1  |     (ecto 3.5.5) lib/ecto/multi.ex:646: Ecto.Multi.apply_operation/5
blockscout_1  | 
blockscout_1  | 2021-12-09T15:35:43.362 application=indexer fetcher=block_catchup first_block_number=14818575 last_block_number=14818575 [error] ** (Postgrex.Error) ERROR 42P10 (invalid_column_reference) there is no unique or exclusion constraint matching the ON CONFLICT specification
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:751: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:660: Ecto.Adapters.SQL.insert_all/8
blockscout_1  |     (ecto 3.5.5) lib/ecto/repo/schema.ex:54: Ecto.Repo.Schema.do_insert_all/6
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/address/token_balances.ex:67: Explorer.Chain.Import.Runner.Address.TokenBalances.insert/3
blockscout_1  |     (ecto 3.5.5) lib/ecto/multi.ex:646: Ecto.Multi.apply_operation/5
blockscout_1  | 
blockscout_1  | 
blockscout_1  | Retrying.
blockscout_1  | 2021-12-09T15:35:43.363 application=explorer fetcher=block_catchup first_block_number=14817863 last_block_number=14817863 import_id=1352876292 transaction_id=1045832391 [error] Could not insert all of chunk into Elixir.Explorer.Chain.Address.TokenBalance using options because of error.
blockscout_1  | 
blockscout_1  | Chunk Size: 1
blockscout_1  | Chunk:
blockscout_1  | 
blockscout_1  | [%{address_hash: %Explorer.Chain.Hash{byte_count: 20, bytes: <<139, 154, 215, 107, 54, 116, 36, 8, 142, 176, 128, 109, 17, 32, 230, 185, 21, 165, 143, 247>>}, block_number: 14817863, inserted_at: ~U[2021-12-09 15:35:43.331664Z], token_contract_address_hash: %Explorer.Chain.Hash{byte_count: 20, bytes: <<130, 120, 49, 82, 192, 174, 220, 95, 36, 32, 229, 220, 77, 214, 208, 14, 44, 210, 171, 87>>}, updated_at: ~U[2021-12-09 15:35:43.331664Z]}]
blockscout_1  | 
blockscout_1  | Options:
blockscout_1  | 
blockscout_1  | [conflict_target: [:address_hash, :token_contract_address_hash, :block_number], on_conflict: #Ecto.Query<from t0 in Explorer.Chain.Address.TokenBalance, where: fragment("EXCLUDED.value IS NOT NULL") and (is_nil(t0.value_fetched_at) or fragment("? < EXCLUDED.value_fetched_at", t0.value_fetched_at)), update: [set: [value: fragment("EXCLUDED.value"), value_fetched_at: fragment("EXCLUDED.value_fetched_at"), inserted_at: fragment("LEAST(EXCLUDED.inserted_at, ?)", t0.inserted_at), updated_at: fragment("GREATEST(EXCLUDED.updated_at, ?)", t0.updated_at)]]>, returning: true, timeout: 60000, timestamps: %{inserted_at: ~U[2021-12-09 15:35:43.331664Z], updated_at: ~U[2021-12-09 15:35:43.331664Z]}]
blockscout_1  | 
blockscout_1  | Exception:
blockscout_1  | 
blockscout_1  | ** (Postgrex.Error) ERROR 42P10 (invalid_column_reference) there is no unique or exclusion constraint matching the ON CONFLICT specification
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:751: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:660: Ecto.Adapters.SQL.insert_all/8
blockscout_1  |     (ecto 3.5.5) lib/ecto/repo/schema.ex:54: Ecto.Repo.Schema.do_insert_all/6
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/address/token_balances.ex:67: Explorer.Chain.Import.Runner.Address.TokenBalances.insert/3
blockscout_1  |     (ecto 3.5.5) lib/ecto/multi.ex:646: Ecto.Multi.apply_operation/5
blockscout_1  | 
blockscout_1  | 2021-12-09T15:35:43.365 application=indexer fetcher=block_catchup first_block_number=14817863 last_block_number=14817863 [error] ** (Postgrex.Error) ERROR 42P10 (invalid_column_reference) there is no unique or exclusion constraint matching the ON CONFLICT specification
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:751: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:660: Ecto.Adapters.SQL.insert_all/8
blockscout_1  |     (ecto 3.5.5) lib/ecto/repo/schema.ex:54: Ecto.Repo.Schema.do_insert_all/6
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/address/token_balances.ex:67: Explorer.Chain.Import.Runner.Address.TokenBalances.insert/3
blockscout_1  |     (ecto 3.5.5) lib/ecto/multi.ex:646: Ecto.Multi.apply_operation/5
blockscout_1  | 
blockscout_1  | 
blockscout_1  | Retrying.
```
### Environment

* Elixir & Erlang/OTP versions (`elixir -version`): `Elixir 1.11.4 (compiled with Erlang/OTP 23)`
* Operating System: `Ubunut  20.04.2 LTS/Docker 20.10.6`
* Blockscout Version/branch: `3.7.3-beta`

Thanks

**Comment by royki** [comment]
Hi @vbaranov , Can you please take a look into this issue. 
Thanks

**Comment by royki** [comment]
Added logs from rds - 
```sql
STATEMENT:  INSERT INTO "address_token_balances" AS a0 ("address_hash","block_number","inserted_at","token_contract_address_hash","updated_at") VALUES ($1,$2,$3,$4,$5),($6,$7,$8,$9,$10) ON CONFLICT ("address_hash","token_contract_address_hash","block_number") DO UPDATE SET "value" = EXCLUDED.value, "value_fetched_at" = EXCLUDED.value_fetched_at, "inserted_at" = LEAST(EXCLUDED.inserted_at, a0."inserted_at"), "updated_at" = GREATEST(EXCLUDED.updated_at, a0."updated_at") WHERE (EXCLUDED.value IS NOT NULL AND ((a0."value_fetched_at" IS NULL) OR a0."value_fetched_at" < EXCLUDED.value_fetched_at)) RETURNING "updated_at","inserted_at","token_contract_address_hash","address_hash","value_fetched_at","block_number","value","id"
```

**Comment by royki** [comment]
Hi @vbaranov 
Here it is
```sql
                                                Table "public.address_token_balances"
           Column            |            Type             | Collation | Nullable |                      Default                       
-----------------------------+-----------------------------+-----------+----------+----------------------------------------------------
 id                          | bigint                      |           | not null | nextval('address_token_balances_id_seq'::regclass)
 address_hash                | bytea                       |           | not null | 
 block_number                | bigint                      |           | not null | 
 token_contract_address_hash | bytea                       |           | not null | 
 value                       | numeric                     |           |          | 
 value_fetched_at            | timestamp without time zone |           |          | 
 inserted_at                 | timestamp without time zone |           | not null | 
 updated_at                  | timestamp without time zone |           | not null | 
 token_id                    | numeric(78,0)               |           |          | 
 token_type                  | character varying(255)      |           |          | 
Indexes:
    "address_token_balances_pkey" PRIMARY KEY, btree (id)
    "address_token_balances_address_hash_token_contract_address_hash" btree (address_hash, token_contract_address_hash, block_number)
    "address_token_balances_block_number_address_hash_index" btree (block_number, address_hash)
    "address_token_balances_token_contract_address_hash_index" btree (token_contract_address_hash)
    "address_token_balances_token_id_index" btree (token_id)
    "fetched_token_balances" UNIQUE, btree (address_hash, token_contract_address_hash, block_number) WHERE token_id IS NULL
    "fetched_token_balances_with_token_id" UNIQUE, btree (address_hash, token_contract_address_hash, token_id, block_number) WHERE token_id IS NOT NULL
    "unfetched_token_balances" UNIQUE, btree (address_hash, token_contract_address_hash, block_number) WHERE value_fetched_at IS NULL AND token_id IS NULL
    "unfetched_token_balances_with_token_id" UNIQUE, btree (address_hash, token_contract_address_hash, token_id, block_number) WHERE value_fetched_at IS NULL AND token_id IS NOT NULL
Foreign-key constraints:
    "address_token_balances_address_hash_fkey" FOREIGN KEY (address_hash) REFERENCES addresses(hash)
    "address_token_balances_token_contract_address_hash_fkey" FOREIGN KEY (token_contract_address_hash) REFERENCES tokens(contract_address_hash)
```

**Comment by vbaranov** [comment]
@royki are you sure that 4.0.0 is properly applied? Can you take a look at this file in your fork https://github.com/blockscout/blockscout/blob/e617c8d780b4ee3a59a93be3c38f61b5dfa0d7c9/apps/explorer/lib/explorer/chain/import/runner/address/token_balances.ex? Does it look absolutely like this one?

**Comment by royki** [comment]
Hi @vbaranov ,
No we reverted to `v3.7.3` as `v4.0.0` was stuck, so we immediately roll back.

**Comment by royki** [comment]
https://github.com/blockscout/blockscout/blob/e617c8d780b4ee3a59a93be3c38f61b5dfa0d7c9/apps/explorer/lib/explorer/chain/import/runner/address/token_balances.ex -> this is for `v4.0.0` right.

**Comment by royki** [comment]
Hi @vbaranov, I accidentally clicked the closed issue button. My bad. Sorry. Reopened.

**Comment by vbaranov** [comment]
I put this issue on hold until you will confirm the state of that source file. From the logs, I see that constraint violation has a different implementation than it is in 4.0.0. Moreover, we do not experience the issue in our instance, where 4.0.0 has been installed.

**Comment by royki** [comment]
Hi @vbaranov,
**PS**: We are using `v3.7.3` on this instance. Once we applied `v4.0.0` but rolled back as it was stuck (due to DB migration, discussed in discord).
Here is the code of `apps/explorer/lib/explorer/chain/import/runner/address/token_balances.ex`
```
defmodule Explorer.Chain.Import.Runner.Address.TokenBalances do
  @moduledoc """
  Bulk imports `t:Explorer.Chain.Address.TokenBalance.t/0`.
  """

  require Ecto.Query

  import Ecto.Query, only: [from: 2]

  alias Ecto.{Changeset, Multi, Repo}
  alias Explorer.Chain.Address.TokenBalance
  alias Explorer.Chain.Import

  @behaviour Import.Runner

  # milliseconds
  @timeout 60_000

  @type imported :: [TokenBalance.t()]

  @impl Import.Runner
  def ecto_schema_module, do: TokenBalance

  @impl Import.Runner
  def option_key, do: :address_token_balances

  @impl Import.Runner
  def imported_table_row do
    %{
      value_type: "[#{ecto_schema_module()}.t()]",
      value_description: "List of `t:#{ecto_schema_module()}.t/0`s"
    }
  end

  @impl Import.Runner
  def run(multi, changes_list, %{timestamps: timestamps} = options) do
    insert_options =
      options
      |> Map.get(option_key(), %{})
      |> Map.take(~w(on_conflict timeout)a)
      |> Map.put_new(:timeout, @timeout)
      |> Map.put(:timestamps, timestamps)

    Multi.run(multi, :address_token_balances, fn repo, _ ->
      insert(repo, changes_list, insert_options)
    end)
  end

  @impl Import.Runner
  def timeout, do: @timeout

  @spec insert(Repo.t(), [map()], %{
          optional(:on_conflict) => Import.Runner.on_conflict(),
          required(:timeout) => timeout(),
          required(:timestamps) => Import.timestamps()
        }) ::
          {:ok, [TokenBalance.t()]}
          | {:error, [Changeset.t()]}
  def insert(repo, changes_list, %{timeout: timeout, timestamps: timestamps} = options) when is_list(changes_list) do
    on_conflict = Map.get_lazy(options, :on_conflict, &default_on_conflict/0)

    # Enforce TokenBalance ShareLocks order (see docs: sharelocks.md)
    ordered_changes_list =
      Enum.sort_by(changes_list, &{&1.token_contract_address_hash, &1.address_hash, &1.block_number})

    {:ok, _} =
      Import.insert_changes_list(
        repo,
        ordered_changes_list,
        conflict_target: ~w(address_hash token_contract_address_hash block_number)a,
        on_conflict: on_conflict,
        for: TokenBalance,
        returning: true,
        timeout: timeout,
        timestamps: timestamps
      )
  end

  defp default_on_conflict do
    from(
      token_balance in TokenBalance,
      update: [
        set: [
          value: fragment("EXCLUDED.value"),
          value_fetched_at: fragment("EXCLUDED.value_fetched_at"),
          inserted_at: fragment("LEAST(EXCLUDED.inserted_at, ?)", token_balance.inserted_at),
          updated_at: fragment("GREATEST(EXCLUDED.updated_at, ?)", token_balance.updated_at)
        ]
      ],
      where:
        fragment("EXCLUDED.value IS NOT NULL") and
          (is_nil(token_balance.value_fetched_at) or
             fragment("? < EXCLUDED.value_fetched_at", token_balance.value_fetched_at))
    )
  end
end
```

**Comment by vbaranov** [comment]
You need to continue updating to 4.0.0 in order to resolve this issue.

**Comment by royki** [comment]
Hi @vbaranov ,
I applied 1st this command - `INSERT INTO schema_migrations VALUES (20211018170638, now());` -> (what is this value **20211018170638** ? From where we get this value?)
Though DB starts responding but after that it is throwing these -
```sql
blockscout_1  | 0x1204700000000000000000000000000000000003@�: (-32602) Unknown block number
blockscout_1  | 0x107cfdadf073ff2bb14631ec91b58a1de3f5b314@�: (-32602) Unknown block number
blockscout_1  | 0xca87e042c28937f7aa48f150536e07ce00dad313@�: (-32602) Unknown block number
blockscout_1  | 0x79f0acf4a8f1281ea2b2e51d7e4a89001706d064@�: (-32602) Unknown block number
blockscout_1  | 0x0264aa3edbc4f074a2a61b9a923e05aa3cfdfb84@�: (-32602) Unknown block number
blockscout_1  | 0xeffa2ac996e3e330a19c6640cf9bb21d06364334@�: (-32602) Unknown block number
blockscout_1  | 0xe1d09dcf03cee4ff96978b47f02d036bc452be16@�: (-32602) Unknown block number
blockscout_1  | 0x1204700000000000000000000000000000000003@�: (-32602) Unknown block number
blockscout_1  | 0x98a91354c81a0571bab1bd3ba4ac344e97f4a868@�: (-32602) Unknown block number
blockscout_1  | 0x6f74b810c1fa4754cd166036a66599e1bbbae3b7@�: (-32602) Unknown block number
blockscout_1  | 0x28b590744f048f054d8c6c28f168cb03b3f1aec3@�: (-32602) Unknown block number
blockscout_1  | 0x77901f14183b1669c80 (truncated)
blockscout_1  | 2021-12-21T12:10:20.063 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: [%{code: -32000, data: %{"blockNumber" => 15059930}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15084485}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15054962}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15066425}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15102603}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15080134}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15034670}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15053762}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15063121}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15070713}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}]
blockscout_1  | 2021-12-21T12:10:20.063 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: [%{code: -32000, data: %{"blockNumber" => 15051296}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15050487}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15075306}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15084076}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15103021}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15050947}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15055943}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15082885}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15085439}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15089637}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}]
blockscout_1  | 2021-12-21T12:10:20.064 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: [%{code: -32000, data: %{"blockNumber" => 15067432}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15052158}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15039952}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15040923}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15050123}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15075174}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15079859}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15044498}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15053589}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15059846}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}]
blockscout_1  | 2021-12-21T12:10:20.095 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: [%{code: -32000, data: %{"blockNumber" => 15055916}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15067817}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15054513}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15089220}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15100342}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15094373}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15071960}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15074274}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15074840}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15037544}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}]
blockscout_1  | 2021-12-21T12:10:20.109 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: [%{code: -32000, data: %{"blockNumber" => 15051845}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15061698}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15100274}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15093513}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15084715}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15048735}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15096436}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15062105}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15052155}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15068099}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}]
blockscout_1  | 2021-12-21T12:10:20.129 application=indexer fetcher=internal_transaction count=10 error_count=10 [error] failed to fetch internal transactions for blocks: [%{code: -32000, data: %{"blockNumber" => 15056861}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15065850}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15084443}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15077962}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15063506}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15042069}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15064444}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15090138}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15080789}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}, %{code: -32000, data: %{"blockNumber" => 15057480}, message: "This request is not supported because your node is running with state pruning. Run with --pruning=archive."}]
blockscout_1  | 2021-12-21T12:10:20.144 application=db_connection [info] Postgrex.Protocol (#PID<0.876.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1050.0> exited
blockscout_1  | 2021-12-21T12:10:20.153 application=db_connection [info] Postgrex.Protocol (#PID<0.871.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1039.0> exited
blockscout_1  | 2021-12-21T12:10:20.160 application=db_connection [info] Postgrex.Protocol (#PID<0.873.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1028.0> exited
blockscout_1  | 2021-12-21T12:10:20.175 application=db_connection [info] Postgrex.Protocol (#PID<0.870.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1011.0> exited
blockscout_1  | 2021-12-21T12:10:20.175 application=db_connection [info] Postgrex.Protocol (#PID<0.874.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1007.0> exited
blockscout_1  | 2021-12-21T12:10:20.177 application=db_connection [info] Postgrex.Protocol (#PID<0.869.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.3136.0> exited
blockscout_1  | 2021-12-21T12:10:20.177 application=db_connection [info] Postgrex.Protocol (#PID<0.865.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.3252.0> exited
blockscout_1  | 2021-12-21T12:10:20.177 application=db_connection [info] Postgrex.Protocol (#PID<0.866.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.1265.0> exited
blockscout_1  | 2021-12-21T12:10:20.178 application=db_connection [info] Postgrex.Protocol (#PID<0.872.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.4176.0> exited
blockscout_1  | 2021-12-21T12:10:20.187 application=db_connection [info] Postgrex.Protocol (#PID<0.875.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.954.0> exited
blockscout_1  | {"Kernel pid terminated",application_controller,"{application_terminated,block_scout_web,shutdown}"}
blockscout_1  | Kernel pid terminated (application_controller) ({application_terminated,block_scout_web,shutdown})
blockscout_1  | 
blockscout_1  | Crash dump is being written to: erl_crash.dump...done
```
- The node is already running with "pruning= archive". The node is still syncing.

- I applied the 2nd command as well - `CREATE INDEX ON public.logs USING btree (address_hash, transaction_hash);`
This command is showing -> **Waiting for the query to complete**

**Comment by vbaranov** [comment]
> The node is already running with "pruning= archive". The node is still syncing. 

The node should be fully synchronized before starting Blockscout on it.

**Comment by royki** [comment]
Thanks @vbaranov . I ll try with the Sync Node. 

What about this one - 
- I applied the 2nd command as well - `CREATE INDEX ON public.logs USING btree (address_hash, transaction_hash);`
This command is showing -> **Waiting for the query to complete**
- And this value - `INSERT INTO schema_migrations VALUES (20211018170638, now());` **20211018170638** -  From where we get this value?

**Comment by vbaranov** [comment]
> I applied the 2nd command as well - CREATE INDEX ON public.logs USING btree (address_hash, transaction_hash);
> This command is showing -> Waiting for the query to complete

You can consider building index concurrently https://www.postgresql.org/docs/9.1/sql-createindex.html

> And this value - INSERT INTO schema_migrations VALUES (20211018170638, now()); 20211018170638 - From where we get this value?

From here `apps/explorer/priv/repo/migrations/20211018170638_add_logs_address_hash_transaction_hash_index.exs` the time when DB migration file has been created.

**Comment by royki** [comment]
Hi @vbaranov ,
Thanks for your reply. 
Even building index concurrently doesn't apply immediately. Is it ok to avoid this indexing ? Or some otherway to apply this? Maybe need to find a way to apply as a background job.

**Comment by royki** [comment]
Hi @vbaranov ,
Is this command ok - `CREATE INDEX CONCURRENTLY ON public.logs USING btree (address_hash, transaction_hash);` .
I'm running this as background job and it is still running, more than 2 days and there is no log.

**Comment by vbaranov** [comment]
@royki yes, the command is fine. `logs` table is the 2nd largest table. You should wait to finish the command. In the meantime, the application can be run on that DB since the DB index is creating concurrently.

**Comment by royki** [comment]
Hi @vbaranov ,
Thanks. One of our explorer instance is successfully updated and running without any issues. 
But there is another instance I followed the same here it s stuck and initially execute some queries which I didn't observe in the last one - 
Here is the blockscout log where it is stuck - 
```sql
blockscout_1  | The database for Explorer.Repo has already been created
blockscout_1  | 
blockscout_1  | 12:52:22.612 [info]  == Running 20211013190346 Explorer.Repo.Migrations.RemoveDuplicatesOfCurrentTokenBalances.change/0 forward
blockscout_1  | 
blockscout_1  | 12:52:22.642 [info]  execute "DELETE FROM address_current_token_balances\nWHERE id in (\n    SELECT a.id FROM (SELECT actb.*\n    FROM address_current_token_balances actb\n    INNER JOIN tokens t\n    ON actb.token_contract_address_hash = t.contract_address_hash\n    WHERE t.type='ERC-721') AS a\nLEFT JOIN\n    (SELECT actb.address_hash, actb.token_contract_address_hash, MAX(actb.value_fetched_at) AS max_value_fetched_at\n    FROM address_current_token_balances actb\n    INNER JOIN tokens t\n    ON actb.token_contract_address_hash = t.contract_address_hash\n    WHERE t.type='ERC-721'\n    GROUP BY token_contract_address_hash, address_hash) c\nON a.address_hash=c.address_hash AND a.token_contract_address_hash = c.token_contract_address_hash AND a.value_fetched_at = c.max_value_fetched_at\nWHERE c.address_hash IS NULL\n);\n"
blockscout_1  | 
blockscout_1  | 12:52:22.852 [info]  execute "UPDATE address_current_token_balances\nSET token_id = NULL\nWHERE id in (\n    SELECT a.id FROM (SELECT actb.*\n    FROM address_current_token_balances actb\n    INNER JOIN tokens t\n    ON actb.token_contract_address_hash = t.contract_address_hash\n    WHERE t.type='ERC-721'\n    AND actb.token_id IS NOT NULL\n    ) a\n);\n"
blockscout_1  | 
blockscout_1  | 12:52:23.045 [info]  execute "UPDATE address_current_token_balances\nSET token_type = t.type\nFROM tokens t\nWHERE address_current_token_balances.token_type IS NULL\nAND t.contract_address_hash = address_current_token_balances.token_contract_address_hash;\n"
blockscout_1  | 
blockscout_1  | 12:53:03.761 [info]  execute "UPDATE address_token_balances\nSET token_type = t.type\nFROM tokens t\nWHERE address_token_balances.token_type IS NULL\nAND t.contract_address_hash = address_token_balances.token_contract_address_hash;\n"
blockscout_1  | 
blockscout_1  | 13:01:54.066 [info]  == Migrated 20211013190346 in 571.3s
blockscout_1  | 
blockscout_1  | 13:01:55.109 [info]  == Running 20211017135545 Explorer.Repo.Migrations.MigrateOptimizationRunsToInt8.up/0 forward
blockscout_1  | 
blockscout_1  | 13:01:55.109 [info]  alter table smart_contracts
blockscout_1  | 
blockscout_1  | 13:01:56.257 [info]  == Migrated 20211017135545 in 1.1s
blockscout_1  | 
blockscout_1  | 13:01:57.043 [info]  == Running 20211018072347 Explorer.Repo.Migrations.AddIsEmptyIndex.change/0 forward
blockscout_1  | 
blockscout_1  | 13:01:57.043 [info]  create index empty_consensus_blocks

```

**Comment by vbaranov** [comment]
@royki congratulations on running one of the instance

> INSERT INTO schema_migrations VALUES (20211013190346, now()); -> 20211013190346

You don't need to run this query in backgorund since this migration already finished `Migrated 20211013190346 in 571.3s`

> INSERT INTO schema_migrations VALUES (20211018072347, now()); -> 20211013190347

DB migrations run consequently. Thus, only the last one is executing at the moment. If `20211018072347` is still executing, you can repeat the same procedure for it:
- skip it with `INSERT INTO schema_migrations VALUES (20211018072347, now());`
- run in the background

**Comment by royki** [comment]
Hi @vbaranov ,
Thanks for your reply. 
I'll retry again. Hope it would work without any glitch.
Can you please tell me, if I execute this query - `INSERT INTO schema_migrations VALUES (20211018072347, now());` , what query I need to run in background?
Is this the corresponding query to run in background - `CREATE INDEX CONCURRENTLY  ON public.blocks USING btree (consensus) WHERE is_empty IS NULL;`

**Comment by vbaranov** [comment]
> Is this the corresponding query to run in background - CREATE INDEX CONCURRENTLY ON public.blocks USING btree (consensus) WHERE is_empty IS NULL;

Yes, right

**Comment by royki** [comment]
Hi @vbaranov ,
It gets stuck 
```sql 
Attaching to docker-stack_blockscout_1
blockscout_1  | ** (Mix) The database for Explorer.Repo couldn't be created: connection not available and request was dropped from queue after 2000ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:
blockscout_1  | 
blockscout_1  |   1. Ensuring your database is available and that you can connect to it
blockscout_1  |   2. Tracking down slow queries and making sure they are running fast enough
blockscout_1  |   3. Increasing the pool_size (albeit it increases resource consumption)
blockscout_1  |   4. Allowing requests to wait longer by increasing :queue_target and :queue_interval
blockscout_1  | 
blockscout_1  | See DBConnection.start_link/2 for more information
blockscout_1  | 
blockscout_1  | The database for Explorer.Repo has already been created
blockscout_1  | 
blockscout_1  | 12:51:31.614 [info]  == Running 20211013190346 Explorer.Repo.Migrations.RemoveDuplicatesOfCurrentTokenBalances.change/0 forward
blockscout_1  | 
blockscout_1  | 12:51:31.621 [info]  execute "DELETE FROM address_current_token_balances\nWHERE id in (\n    SELECT a.id FROM (SELECT actb.*\n    FROM address_current_token_balances actb\n    INNER JOIN tokens t\n    ON actb.token_contract_address_hash = t.contract_address_hash\n    WHERE t.type='ERC-721') AS a\nLEFT JOIN\n    (SELECT actb.address_hash, actb.token_contract_address_hash, MAX(actb.value_fetched_at) AS max_value_fetched_at\n    FROM address_current_token_balances actb\n    INNER JOIN tokens t\n    ON actb.token_contract_address_hash = t.contract_address_hash\n    WHERE t.type='ERC-721'\n    GROUP BY token_contract_address_hash, address_hash) c\nON a.address_hash=c.address_hash AND a.token_contract_address_hash = c.token_contract_address_hash AND a.value_fetched_at = c.max_value_fetched_at\nWHERE c.address_hash IS NULL\n);\n"
blockscout_1  | 
blockscout_1  | 12:51:34.153 [info]  execute "UPDATE address_current_token_balances\nSET token_id = NULL\nWHERE id in (\n    SELECT a.id FROM (SELECT actb.*\n    FROM address_current_token_balances actb\n    INNER JOIN tokens t\n    ON actb.token_contract_address_hash = t.contract_address_hash\n    WHERE t.type='ERC-721'\n    AND actb.token_id IS NOT NULL\n    ) a\n);\n"
blockscout_1  | 
blockscout_1  | 12:51:34.162 [info]  execute "UPDATE address_current_token_balances\nSET token_type = t.type\nFROM tokens t\nWHERE address_current_token_balances.token_type IS NULL\nAND t.contract_address_hash = address_current_token_balances.token_contract_address_hash;\n"
blockscout_1  | 
blockscout_1  | 12:53:31.409 [info]  execute "UPDATE address_token_balances\nSET token_type = t.type\nFROM tokens t\nWHERE address_token_balances.token_type IS NULL\nAND t.contract_address_hash = address_token_balances.token_contract_address_hash;\n"
```

**Comment by royki** [comment]
Hi @vbaranov, 
It is fine now. I just waited few mins and it s migrated all. Now it is working.

But one thing has not been yet faded away from the 1st instance - 
```sql
blockscout_1  | Exception:
blockscout_1  | 
blockscout_1  | ** (Postgrex.Error) ERROR 23503 (foreign_key_violation) insert or update on table "internal_transactions" violates foreign key constraint "internal_transactions_transaction_hash_fkey"
blockscout_1  | 
blockscout_1  |     table: internal_transactions
blockscout_1  |     constraint: internal_transactions_transaction_hash_fkey
blockscout_1  | 
blockscout_1  | Key (transaction_hash)=(\x6a7bc4d908beecaab4af16f33dafd50496eb3304500fc8aacc221c69f2efd076) is not present in table "transactions".
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:756: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:663: Ecto.Adapters.SQL.insert_all/9
blockscout_1  |     (ecto 3.7.1) lib/ecto/repo/schema.ex:58: Ecto.Repo.Schema.do_insert_all/7
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/internal_transactions.ex:152: Explorer.Chain.Import.Runner.InternalTransactions.insert/3
blockscout_1  |     (ecto 3.7.1) lib/ecto/multi.ex:716: Ecto.Multi.apply_operation/5
blockscout_1  | 
blockscout_1  | 2022-01-07T14:27:29.920 fetcher=internal_transaction count=1 [error] Task #PID<0.1833.235> started from Indexer.Fetcher.InternalTransaction terminating
blockscout_1  | ** (Postgrex.Error) ERROR 23503 (foreign_key_violation) insert or update on table "internal_transactions" violates foreign key constraint "internal_transactions_transaction_hash_fkey"
blockscout_1  | 
blockscout_1  |     table: internal_transactions
blockscout_1  |     constraint: internal_transactions_transaction_hash_fkey
blockscout_1  | 
blockscout_1  | Key (transaction_hash)=(\x6a7bc4d908beecaab4af16f33dafd50496eb3304500fc8aacc221c69f2efd076) is not present in table "transactions".
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:756: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:663: Ecto.Adapters.SQL.insert_all/9
blockscout_1  |     (ecto 3.7.1) lib/ecto/repo/schema.ex:58: Ecto.Repo.Schema.do_insert_all/7
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/internal_transactions.ex:152: Explorer.Chain.Import.Runner.InternalTransactions.insert/3
blockscout_1  |     (ecto 3.7.1) lib/ecto/multi.ex:716: Ecto.Multi.apply_operation/5
blockscout_1  | Function: &Indexer.BufferedTask.log_run/1
blockscout_1  |     Args: [%{batch: [14915994], callback_module: Indexer.Fetcher.InternalTransaction, callback_module_state: [transport: EthereumJSONRPC.HTTP, transport_options: [http: EthereumJSONRPC.HTTP.HTTPoison, url: "http://parity:8545", method_to_url: [eth_getBalance: "http://parity:8545", trace_block: "http://parity:8545", trace_replayTransaction: "http://parity:8545"], http_options: [recv_timeout: 600000, timeout: 600000, hackney: [pool: :ethereum_jsonrpc]]], variant: EthereumJSONRPC.Parity], metadata: [fetcher: :internal_transaction]}]
```

- I have noticed some changes in the template. Apart from that the in the footer, the icons are not visible in the home page. If we navigate to other transaction or block page, the icons in the footer are visible.
![image](https://user-images.githubusercontent.com/1545990/148558389-c68c309e-036f-42ae-8baa-5b174faf4113.png)

![image](https://user-images.githubusercontent.com/1545990/148558413-d84d60f5-a582-4a82-b98f-7a8055928ee7.png)

**Comment by vbaranov** [comment]
> But one thing has not been yet faded away from the 1st instance -

Most likely it is a bug of archive node: `trace_replayBlockTransactions` returns a hash of transaction (`0x6a7bc4d908beecaab4af16f33dafd50496eb3304500fc8aacc221c69f2efd076`), which doesn't exist in the block with number `14915994`.

> I have noticed some changes in the template. Apart from that the in the footer, the icons are not visible in the home page. If we navigate to other transaction or block page, the icons in the footer are visible.

This extended set of icons is customization made in your instance.


I am closing this issue.

**Comment by royki** [comment]
Hi @vbaranov ,

Following the discord conversation, here is the ticket that was created - 
We are still getting following errors - 

```sql
blockscout_1  | [conflict_target: [:block_hash, :block_index], on_conflict: #Ecto.Query<from i0 in Explorer.Chain.InternalTransaction, where: fragment("(EXCLUDED.transaction_hash, EXCLUDED.index, EXCLUDED.call_type, EXCLUDED.created_contract_address_hash, EXCLUDED.created_contract_code, EXCLUDED.error, EXCLUDED.from_address_hash, EXCLUDED.gas, EXCLUDED.gas_used, EXCLUDED.init, EXCLUDED.input, EXCLUDED.output, EXCLUDED.to_address_hash, EXCLUDED.trace_address, EXCLUDED.transaction_index, EXCLUDED.type, EXCLUDED.value) IS DISTINCT FROM (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)", i0.transaction_hash, i0.index, i0.call_type, i0.created_contract_address_hash, i0.created_contract_code, i0.error, i0.from_address_hash, i0.gas, i0.gas_used, i0.init, i0.input, i0.output, i0.to_address_hash, i0.trace_address, i0.transaction_index, i0.type, i0.value), update: [set: [block_number: fragment("EXCLUDED.block_number"), call_type: fragment("EXCLUDED.call_type"), created_contract_address_hash: fragment("EXCLUDED.created_contract_address_hash"), created_contract_code: fragment("EXCLUDED.created_contract_code"), error: fragment("EXCLUDED.error"), from_address_hash: fragment("EXCLUDED.from_address_hash"), gas: fragment("EXCLUDED.gas"), gas_used: fragment("EXCLUDED.gas_used"), index: fragment("EXCLUDED.index"), init: fragment("EXCLUDED.init"), input: fragment("EXCLUDED.input"), output: fragment("EXCLUDED.output"), to_address_hash: fragment("EXCLUDED.to_address_hash"), trace_address: fragment("EXCLUDED.trace_address"), transaction_hash: fragment("EXCLUDED.transaction_hash"), transaction_index: fragment("EXCLUDED.transaction_index"), type: fragment("EXCLUDED.type"), value: fragment("EXCLUDED.value"), inserted_at: fragment("LEAST(?, EXCLUDED.inserted_at)", i0.inserted_at), updated_at: fragment("GREATEST(?, EXCLUDED.updated_at)", i0.updated_at)]]>, returning: true, timeout: 60000, timestamps: %{inserted_at: ~U[2022-02-10 11:23:20.349948Z], updated_at: ~U[2022-02-10 11:23:20.349948Z]}]
blockscout_1  | 
blockscout_1  | Exception:
blockscout_1  | 
blockscout_1  | ** (Postgrex.Error) ERROR 23503 (foreign_key_violation) insert or update on table "internal_transactions" violates foreign key constraint "internal_transactions_transaction_hash_fkey"
blockscout_1  | 
blockscout_1  |     table: internal_transactions
blockscout_1  |     constraint: internal_transactions_transaction_hash_fkey
blockscout_1  | 
blockscout_1  | Key (transaction_hash)=(\x5d6816a9e8138e39fb88b6158e27683c44771f973fca9939d507b5674d41be42) is not present in table "transactions".
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:756: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:663: Ecto.Adapters.SQL.insert_all/9
blockscout_1  |     (ecto 3.7.1) lib/ecto/repo/schema.ex:58: Ecto.Repo.Schema.do_insert_all/7
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/internal_transactions.ex:152: Explorer.Chain.Import.Runner.InternalTransactions.insert/3
blockscout_1  |     (ecto 3.7.1) lib/ecto/multi.ex:716: Ecto.Multi.apply_operation/5
blockscout_1  | 
blockscout_1  | 2022-02-10T11:23:20.380 fetcher=internal_transaction count=10 [error] Task #PID<0.18646.553> started from Indexer.Fetcher.InternalTransaction terminating
blockscout_1  | ** (Postgrex.Error) ERROR 23503 (foreign_key_violation) insert or update on table "internal_transactions" violates foreign key constraint "internal_transactions_transaction_hash_fkey"
blockscout_1  | 
blockscout_1  |     table: internal_transactions
blockscout_1  |     constraint: internal_transactions_transaction_hash_fkey
blockscout_1  | 
blockscout_1  | Key (transaction_hash)=(\x5d6816a9e8138e39fb88b6158e27683c44771f973fca9939d507b5674d41be42) is not present in table "transactions".
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:756: Ecto.Adapters.SQL.raise_sql_call_error/1
blockscout_1  |     (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:663: Ecto.Adapters.SQL.insert_all/9
blockscout_1  |     (ecto 3.7.1) lib/ecto/repo/schema.ex:58: Ecto.Repo.Schema.do_insert_all/7
blockscout_1  |     (explorer 0.0.1) lib/explorer/repo.ex:45: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
blockscout_1  |     (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import.ex:299: Explorer.Chain.Import.insert_changes_list/3
blockscout_1  |     (explorer 0.0.1) lib/explorer/chain/import/runner/internal_transactions.ex:152: Explorer.Chain.Import.Runner.InternalTransactions.insert/3
blockscout_1  |     (ecto 3.7.1) lib/ecto/multi.ex:716: Ecto.Multi.apply_operation/5
blockscout_1  | Function: &Indexer.BufferedTask.log_run/1
blockscout_1  |     Args: [%{batch: [16309254, 16308718, 16306154, 15706768, 16308687, 16307733, 16308096, 16308811, 16308785, 16308709], callback_module: Indexer.Fetcher.InternalTransaction, callback_module_state: [transport: EthereumJSONRPC.HTTP, transport_options: [http: EthereumJSONRPC.HTTP.HTTPoison, url: "http://parity:8545", method_to_url: [eth_getBalance: "http://parity:8545", trace_block: "http://parity:8545", trace_replayTransaction: "htt
```
[blockscout.log](https://github.com/blockscout/blockscout/files/8040612/blockscout.log)


I added more logs in the attached file.
